From bfe0cd7dedaaa8f47f4bdbf350c7b1e1df3b770c Mon Sep 17 00:00:00 2001
From: Shubh <shubhaggarwalva@gmail.com>
Date: Tue, 15 Sep 2026 12:57:26 +0530
Subject: [PATCH] JusBrowse: remove alternative launcher icons and fix TabStrip
 firefox logo
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

- AndroidManifest: remove 22 activity-alias entries for alternative icons
  (AppSolid*, AppGradient*, AppRetro etc.) – keep only default App +
  Gecko aliases, ponytail single logo (SVG+PNG)
- TabStrip.kt: ic_firefox -> ic_launcher_foreground (was deleted)
---
 .../fenix/app/src/main/AndroidManifest.xml    | 379 +-----------------
 .../fenix/browser/tabstrip/TabStrip.kt        |   2 +-
 2 files changed, 2 insertions(+), 379 deletions(-)

diff --git a/mobile/android/fenix/app/src/main/AndroidManifest.xml b/mobile/android/fenix/app/src/main/AndroidManifest.xml
index 97f31394d723..57ded2c68516 100644
--- a/mobile/android/fenix/app/src/main/AndroidManifest.xml
+++ b/mobile/android/fenix/app/src/main/AndroidManifest.xml
@@ -104,384 +104,7 @@
                 android:resource="@xml/shortcuts" />
         </activity-alias>
 
-        <activity-alias
-            android:name="${applicationId}.AlternativeApp"
-            android:enabled="false"
-            android:exported="true"
-            android:icon="@mipmap/ic_launcher_alternative"
-            android:roundIcon="@mipmap/ic_launcher_round_alternative"
-            android:targetActivity="HomeActivity">
-            <intent-filter>
-                <action android:name="android.intent.action.MAIN" />
-                <category android:name="android.intent.category.LAUNCHER" />
-            </intent-filter>
-            <meta-data
-                android:name="android.app.shortcuts"
-                android:resource="@xml/shortcuts" />
-        </activity-alias>
-
-        <!--
-            Activity-alias entries used by the app icon selection feature.
-            Only one is enabled at a time to switch the launcher icon dynamically.
-        -->
-        <!-- Solid color icons -->
-        <!-- Alternative icons with different solid background colors. -->
-        <activity-alias
-            android:name="${applicationId}.AppSolidLight"
-            android:enabled="false"
-            android:exported="true"
-            android:icon="@mipmap/ic_launcher_solid_light"
-            android:roundIcon="@mipmap/ic_launcher_solid_light"
-            android:targetActivity="HomeActivity">
-            <intent-filter>
-                <action android:name="android.intent.action.MAIN" />
-                <category android:name="android.intent.category.LAUNCHER" />
-            </intent-filter>
-            <meta-data
-                android:name="android.app.shortcuts"
-                android:resource="@xml/shortcuts" />
-        </activity-alias>
-
-        <activity-alias
-            android:name="${applicationId}.AppSolidDark"
-            android:enabled="false"
-            android:exported="true"
-            android:icon="@mipmap/ic_launcher_solid_dark"
-            android:roundIcon="@mipmap/ic_launcher_solid_dark"
-            android:targetActivity="HomeActivity">
-            <intent-filter>
-                <action android:name="android.intent.action.MAIN" />
-                <category android:name="android.intent.category.LAUNCHER" />
-            </intent-filter>
-            <meta-data
-                android:name="android.app.shortcuts"
-                android:resource="@xml/shortcuts" />
-        </activity-alias>
-
-        <activity-alias
-            android:name="${applicationId}.AppSolidRed"
-            android:enabled="false"
-            android:exported="true"
-            android:icon="@mipmap/ic_launcher_solid_red"
-            android:roundIcon="@mipmap/ic_launcher_solid_red"
-            android:targetActivity="HomeActivity">
-            <intent-filter>
-                <action android:name="android.intent.action.MAIN" />
-                <category android:name="android.intent.category.LAUNCHER" />
-            </intent-filter>
-            <meta-data
-                android:name="android.app.shortcuts"
-                android:resource="@xml/shortcuts" />
-        </activity-alias>
-
-        <activity-alias
-            android:name="${applicationId}.AppSolidGreen"
-            android:enabled="false"
-            android:exported="true"
-            android:icon="@mipmap/ic_launcher_solid_green"
-            android:roundIcon="@mipmap/ic_launcher_solid_green"
-            android:targetActivity="HomeActivity">
-            <intent-filter>
-                <action android:name="android.intent.action.MAIN" />
-                <category android:name="android.intent.category.LAUNCHER" />
-            </intent-filter>
-            <meta-data
-                android:name="android.app.shortcuts"
-                android:resource="@xml/shortcuts" />
-        </activity-alias>
-
-        <activity-alias
-            android:name="${applicationId}.AppSolidBlue"
-            android:enabled="false"
-            android:exported="true"
-            android:icon="@mipmap/ic_launcher_solid_blue"
-            android:roundIcon="@mipmap/ic_launcher_solid_blue"
-            android:targetActivity="HomeActivity">
-            <intent-filter>
-                <action android:name="android.intent.action.MAIN" />
-                <category android:name="android.intent.category.LAUNCHER" />
-            </intent-filter>
-            <meta-data
-                android:name="android.app.shortcuts"
-                android:resource="@xml/shortcuts" />
-        </activity-alias>
-
-        <activity-alias
-            android:name="${applicationId}.AppSolidPurple"
-            android:enabled="false"
-            android:exported="true"
-            android:icon="@mipmap/ic_launcher_solid_purple"
-            android:roundIcon="@mipmap/ic_launcher_solid_purple"
-            android:targetActivity="HomeActivity">
-            <intent-filter>
-                <action android:name="android.intent.action.MAIN" />
-                <category android:name="android.intent.category.LAUNCHER" />
-            </intent-filter>
-            <meta-data
-                android:name="android.app.shortcuts"
-                android:resource="@xml/shortcuts" />
-        </activity-alias>
-
-        <activity-alias
-            android:name="${applicationId}.AppSolidPurpleDark"
-            android:enabled="false"
-            android:exported="true"
-            android:icon="@mipmap/ic_launcher_solid_purple_dark"
-            android:roundIcon="@mipmap/ic_launcher_solid_purple_dark"
-            android:targetActivity="HomeActivity">
-            <intent-filter>
-                <action android:name="android.intent.action.MAIN" />
-                <category android:name="android.intent.category.LAUNCHER" />
-            </intent-filter>
-            <meta-data
-                android:name="android.app.shortcuts"
-                android:resource="@xml/shortcuts" />
-        </activity-alias>
-
-        <!-- Gradient icons -->
-        <!-- Alternative icons with gradient-style background colors. -->
-        <activity-alias
-            android:name="${applicationId}.AppGradientSunrise"
-            android:enabled="false"
-            android:exported="true"
-            android:icon="@mipmap/ic_launcher_gradient_sunrise"
-            android:roundIcon="@mipmap/ic_launcher_gradient_sunrise"
-            android:targetActivity="HomeActivity">
-            <intent-filter>
-                <action android:name="android.intent.action.MAIN" />
-                <category android:name="android.intent.category.LAUNCHER" />
-            </intent-filter>
-            <meta-data
-                android:name="android.app.shortcuts"
-                android:resource="@xml/shortcuts" />
-        </activity-alias>
-
-        <activity-alias
-            android:name="${applicationId}.AppGradientGoldenHour"
-            android:enabled="false"
-            android:exported="true"
-            android:icon="@mipmap/ic_launcher_gradient_golden_hour"
-            android:roundIcon="@mipmap/ic_launcher_gradient_golden_hour"
-            android:targetActivity="HomeActivity">
-            <intent-filter>
-                <action android:name="android.intent.action.MAIN" />
-                <category android:name="android.intent.category.LAUNCHER" />
-            </intent-filter>
-            <meta-data
-                android:name="android.app.shortcuts"
-                android:resource="@xml/shortcuts" />
-        </activity-alias>
-
-        <activity-alias
-            android:name="${applicationId}.AppGradientSunset"
-            android:enabled="false"
-            android:exported="true"
-            android:icon="@mipmap/ic_launcher_gradient_sunset"
-            android:roundIcon="@mipmap/ic_launcher_gradient_sunset"
-            android:targetActivity="HomeActivity">
-            <intent-filter>
-                <action android:name="android.intent.action.MAIN" />
-                <category android:name="android.intent.category.LAUNCHER" />
-            </intent-filter>
-            <meta-data
-                android:name="android.app.shortcuts"
-                android:resource="@xml/shortcuts" />
-        </activity-alias>
-
-        <activity-alias
-            android:name="${applicationId}.AppGradientBlueHour"
-            android:enabled="false"
-            android:exported="true"
-            android:icon="@mipmap/ic_launcher_gradient_blue_hour"
-            android:roundIcon="@mipmap/ic_launcher_gradient_blue_hour"
-            android:targetActivity="HomeActivity">
-            <intent-filter>
-                <action android:name="android.intent.action.MAIN" />
-                <category android:name="android.intent.category.LAUNCHER" />
-            </intent-filter>
-            <meta-data
-                android:name="android.app.shortcuts"
-                android:resource="@xml/shortcuts" />
-        </activity-alias>
-
-        <activity-alias
-            android:name="${applicationId}.AppGradientTwilight"
-            android:enabled="false"
-            android:exported="true"
-            android:icon="@mipmap/ic_launcher_gradient_twilight"
-            android:roundIcon="@mipmap/ic_launcher_gradient_twilight"
-            android:targetActivity="HomeActivity">
-            <intent-filter>
-                <action android:name="android.intent.action.MAIN" />
-                <category android:name="android.intent.category.LAUNCHER" />
-            </intent-filter>
-            <meta-data
-                android:name="android.app.shortcuts"
-                android:resource="@xml/shortcuts" />
-        </activity-alias>
-
-        <activity-alias
-            android:name="${applicationId}.AppGradientMidnight"
-            android:enabled="false"
-            android:exported="true"
-            android:icon="@mipmap/ic_launcher_gradient_midnight"
-            android:roundIcon="@mipmap/ic_launcher_gradient_midnight"
-            android:targetActivity="HomeActivity">
-            <intent-filter>
-                <action android:name="android.intent.action.MAIN" />
-                <category android:name="android.intent.category.LAUNCHER" />
-            </intent-filter>
-            <meta-data
-                android:name="android.app.shortcuts"
-                android:resource="@xml/shortcuts" />
-        </activity-alias>
-
-        <activity-alias
-            android:name="${applicationId}.AppGradientNorthernLights"
-            android:enabled="false"
-            android:exported="true"
-            android:icon="@mipmap/ic_launcher_gradient_northern_lights"
-            android:roundIcon="@mipmap/ic_launcher_gradient_northern_lights"
-            android:targetActivity="HomeActivity">
-            <intent-filter>
-                <action android:name="android.intent.action.MAIN" />
-                <category android:name="android.intent.category.LAUNCHER" />
-            </intent-filter>
-            <meta-data
-                android:name="android.app.shortcuts"
-                android:resource="@xml/shortcuts" />
-        </activity-alias>
-
-        <!-- Themed/alternative icons -->
-        <!-- Icons with unique designs or special themes (e.g. retro, pixelated, Pride). -->
-        <activity-alias
-            android:name="${applicationId}.AppRetro2004"
-            android:enabled="false"
-            android:exported="true"
-            android:icon="@mipmap/ic_launcher_retro_2004"
-            android:roundIcon="@mipmap/ic_launcher_retro_2004"
-            android:targetActivity="HomeActivity">
-            <intent-filter>
-                <action android:name="android.intent.action.MAIN" />
-                <category android:name="android.intent.category.LAUNCHER" />
-            </intent-filter>
-            <meta-data
-                android:name="android.app.shortcuts"
-                android:resource="@xml/shortcuts" />
-        </activity-alias>
-
-        <activity-alias
-            android:name="${applicationId}.AppPixelated"
-            android:enabled="false"
-            android:exported="true"
-            android:icon="@mipmap/ic_launcher_pixelated"
-            android:roundIcon="@mipmap/ic_launcher_pixelated"
-            android:targetActivity="HomeActivity">
-            <intent-filter>
-                <action android:name="android.intent.action.MAIN" />
-                <category android:name="android.intent.category.LAUNCHER" />
-            </intent-filter>
-            <meta-data
-                android:name="android.app.shortcuts"
-                android:resource="@xml/shortcuts" />
-        </activity-alias>
-
-        <activity-alias
-            android:name="${applicationId}.AppCuddling"
-            android:enabled="false"
-            android:exported="true"
-            android:icon="@mipmap/ic_launcher_cuddling"
-            android:roundIcon="@mipmap/ic_launcher_cuddling"
-            android:targetActivity="HomeActivity">
-            <intent-filter>
-                <action android:name="android.intent.action.MAIN" />
-                <category android:name="android.intent.category.LAUNCHER" />
-            </intent-filter>
-            <meta-data
-                android:name="android.app.shortcuts"
-                android:resource="@xml/shortcuts" />
-        </activity-alias>
-
-        <activity-alias
-            android:name="${applicationId}.AppPride"
-            android:enabled="false"
-            android:exported="true"
-            android:icon="@mipmap/ic_launcher_pride"
-            android:roundIcon="@mipmap/ic_launcher_pride"
-            android:targetActivity="HomeActivity">
-            <intent-filter>
-                <action android:name="android.intent.action.MAIN" />
-                <category android:name="android.intent.category.LAUNCHER" />
-            </intent-filter>
-            <meta-data
-                android:name="android.app.shortcuts"
-                android:resource="@xml/shortcuts" />
-        </activity-alias>
-
-        <activity-alias
-            android:name="${applicationId}.AppFlaming"
-            android:enabled="false"
-            android:exported="true"
-            android:icon="@mipmap/ic_launcher_flaming"
-            android:roundIcon="@mipmap/ic_launcher_flaming"
-            android:targetActivity="HomeActivity">
-            <intent-filter>
-                <action android:name="android.intent.action.MAIN" />
-                <category android:name="android.intent.category.LAUNCHER" />
-            </intent-filter>
-            <meta-data
-                android:name="android.app.shortcuts"
-                android:resource="@xml/shortcuts" />
-        </activity-alias>
-
-        <activity-alias
-            android:name="${applicationId}.AppMinimal"
-            android:enabled="false"
-            android:exported="true"
-            android:icon="@mipmap/ic_launcher_minimal"
-            android:roundIcon="@mipmap/ic_launcher_minimal"
-            android:targetActivity="HomeActivity">
-            <intent-filter>
-                <action android:name="android.intent.action.MAIN" />
-                <category android:name="android.intent.category.LAUNCHER" />
-            </intent-filter>
-            <meta-data
-                android:name="android.app.shortcuts"
-                android:resource="@xml/shortcuts" />
-        </activity-alias>
-
-        <activity-alias
-            android:name="${applicationId}.AppMomo"
-            android:enabled="false"
-            android:exported="true"
-            android:icon="@mipmap/ic_launcher_momo"
-            android:roundIcon="@mipmap/ic_launcher_momo"
-            android:targetActivity="HomeActivity">
-            <intent-filter>
-                <action android:name="android.intent.action.MAIN" />
-                <category android:name="android.intent.category.LAUNCHER" />
-            </intent-filter>
-            <meta-data
-                android:name="android.app.shortcuts"
-                android:resource="@xml/shortcuts" />
-        </activity-alias>
-        <activity-alias
-            android:name="${applicationId}.AppCool"
-            android:enabled="false"
-            android:exported="true"
-            android:icon="@mipmap/ic_launcher_cool"
-            android:roundIcon="@mipmap/ic_launcher_cool"
-            android:targetActivity="HomeActivity">
-            <intent-filter>
-                <action android:name="android.intent.action.MAIN" />
-                <category android:name="android.intent.category.LAUNCHER" />
-            </intent-filter>
-            <meta-data
-                android:name="android.app.shortcuts"
-                android:resource="@xml/shortcuts" />
-        </activity-alias>
-
-        <!--
+        <!-- JusBrowse: single logo only (SVG+PNG), alternative icons removed -->
           Fennec declared entry for homescreen pinned shortcuts.
         -->
         <activity-alias
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/browser/tabstrip/TabStrip.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/browser/tabstrip/TabStrip.kt
index 85c262422017..ce60a7527869 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/browser/tabstrip/TabStrip.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/browser/tabstrip/TabStrip.kt
@@ -478,7 +478,7 @@ private fun TabStripIcon(
             )
         } else if (url == ABOUT_HOME_URL) {
             Favicon(
-                imageResource = R.drawable.ic_firefox,
+                imageResource = R.drawable.ic_launcher_foreground,
                 size = tabStripIconSize,
             )
         } else {
-- 
2.55.0

